src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -0,0 +1,103 @@
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Number.class, (left, right) -> left.doubleValue() / right.doubleValue());
+		Arithmetics.registerOperation(Operator.EXPONENTIATION, Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right) && right.longValue() >= 0)
+				return (long) Math.pow(left.longValue(), right.longValue());
+			return Math.pow(left.doubleValue(), right.doubleValue());
+		});
+		Arithmetics.registerDifference(Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return Math.abs(left.longValue() - right.longValue());
+			return Math.abs(left.doubleValue() - right.doubleValue());
+		});
+		Arithmetics.registerDefaultValue(Number.class, () -> 0L);
+
+		// Vector - Vector
+		Arithmetics.registerOperation(Operator.ADDITION, Vector.class, (left, right) -> left.clone().add(right));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Vector.class, (left, right) -> left.clone().subtract(right));
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, (left, right) -> left.clone().multiply(right));
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, (left, right) -> left.clone().divide(right));
+
+		// Vector - Number
+		// Number - Vector
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, Number.class, (left, right) -> left.clone().multiply(right.doubleValue()), (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.multiply(right);
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, Number.class, (left, right) -> {
+			double number = right.doubleValue();
+			Vector rightVector = new Vector(number, number, number);
+			return left.clone().divide(rightVector);
+		}, (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.divide(right);
+		});
+		Arithmetics.registerDifference(Vector.class,
+			(left, right) -> new Vector(Math.abs(left.getX() - right.getX()), Math.abs(left.getY() - right.getY()), Math.abs(left.getZ() - right.getZ())));
+		Arithmetics.registerDefaultValue(Vector.class, Vector::new);
+
+		// Timespan - Timespan
+		Arithmetics.registerOperation(Operator.ADDITION, Timespan.class, (left, right) -> new Timespan(left.getMilliSeconds() + right.getMilliSeconds()));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Timespan.class, (left, right) -> new Timespan(Math.max(0, left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDifference(Timespan.class, (left, right) -> new Timespan(Math.abs(left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDefaultValue(Timespan.class, Timespan::new);
+
+		// Timespan - Number
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Timespan.class, Number.class, (left, right) -> new Timespan(left.getMilliSeconds() * right.longValue()));
```

Comment:
This one should be commutative, `2 * 2 seconds` should equal `4 seconds`, not `0`.